gpytorch/lazy/lazy_tensor.py

```diff
@@ -1329,6 +1330,45 @@ def requires_grad_(self, val):
         self.requires_grad = val
         return self
 
+    @cached(name="diagonalization")
+    def diagonalization(self, method: Optional[str] = None):
+        """
+        Returns a (usually partial) diagonalization of a symmetric PSD matrix.
+        Options are either "lanczos" or "symeig". "lanczos" runs Lanczos while
+        "symeig" runs torch.symeig.
```

Comment:
Maybe clarify that this doesn't necessarily mean calling symeig on the full dense matrix.
```suggestion
        "symeig" runs LazyTensor.symeig.
```